web/source/kmwcallback.js

```diff
@@ -533,14 +527,18 @@ if(!window['keyman']['initialized']) {
         if(Ldoc.body.isContentEditable || Ldoc.designMode.toLowerCase()=='on') { // I1295 - isContentEditable
           Lrange.moveStart('character',-s.length);
           
-          keymanweb._CacheCommandsReset(Ldoc, _CacheableCommands,Lrange.select);
+          kbdInterface._CacheCommandsReset(Ldoc, _CacheableCommands,Lrange.select);
           Lrange.moveStart('character',s.length);
           Lrange.select();
         }
         // Adjust deadkey positions 
         if(dn >= 0) {
-          keymanweb._DeadkeyDeleteMatched();                                  // I3318
-          keymanweb._DeadkeyAdjustPos(LselectionStart, -dn + s._kmwLength()); // I3318
+          // Pelem.selectionStart seems to exist here in IE 9 and is valid... but is this the right approach?
+          // I'm probably missing something or other.
+          var LselectionStart = Pelem.selectionStart;
```

Comment:
Cross-reference with https://github.com/sillsdev/keyman-issues/issues/3309 - a Closure warning appeared here after the refactor that needs proper resolution.  This or removing the `_DeadkeyAdjustPos` beneath it outright seem the most likely two candidates, but I feel like I'm probably missing something for proper resolution here.

Without the line above, `LselectionStart` is undefined for the line below; this was an issue with the original, pre-refactor code as well, but somehow Closure failed to notice.